x-pack/filebeat/module/zeek/files/ingest/pipeline.yml

```diff
@@ -0,0 +1,60 @@
+description: Pipeline for normalizing Zeek files.log
+processors:
+- set:
+    field: event.created
+    value: '{{_ingest.timestamp}}'
+- date:
+    field: zeek.files.ts
+    formats:
+    - UNIX
+- remove:
+    field: zeek.files.ts
+- script:
+    lang: painless
+    source: ctx.zeek.session_id = ctx.zeek.files.session_ids[0];
+    if: ctx.zeek.files.session_ids != null
+    ignore_failure: true
+- script:
+    lang: painless
+    source: ctx.zeek.files.rx_host = ctx.zeek.files.rx_hosts[0]; ctx.zeek.files.remove('rx_hosts');
+    ignore_failure: true
+- script:
+    lang: painless
+    source: ctx.zeek.files.tx_host = ctx.zeek.files.tx_hosts[0]; ctx.zeek.files.remove('tx_hosts');
+    ignore_failure: true
+- set:
+    field: event.id
+    value: '{{zeek.session_id}}'
+    if: ctx.zeek.session_id != null
+- append:
+    field: server.ip
+    value: "{{zeek.files.tx_host}}"
+    if: "ctx?.zeek?.files?.tx_host != null"    
+- append:
+    field: related.ip
+    value: "{{zeek.files.tx_host}}"
+    if: "ctx?.zeek?.files?.tx_host != null"
+- append:
+    field: client.ip
```

Comment:
Should this be a `set`? I thought client.ip was not an array. Same for `server.ip`.

Reply:
yeah, I was waffling on this.  current ECS doc for client.ip is:
```
IP address of the client.
Can be one or multiple IPv4 or IPv6 addresses.
type: ip
```
@webmat thoughts?

Reply:
Oh no, Lee you found another hidden landmine in ECS!

I agree with Andrew, I think this should be a single value. This was discussed here recently https://github.com/elastic/ecs/issues/786 as well.

Did you have a concrete use for this being an array?

Reply:
zeek files has transmit & receive hosts that can be arrays.
[zeek files doc](https://docs.zeek.org/en/current/scripts/base/frameworks/files/main.zeek.html#type-Files::Info)

currently we set the first element of the tx_host to server.ip & first element of rx_host to client.ip.

I'm good with keeping it one value.  I'll just make sure all the tx_hosts & rx_hosts get set in related.ip